code/game/gamemodes/miniantags/abduction/gland.dm

```diff
@@ -14,28 +14,69 @@
 	var/active = 0
 	tough = TRUE //not easily broken by combat damage
 
+	var/mind_control_uses = 1
+	var/mind_control_duration = 1800
+	var/active_mind_control = FALSE
+
 /obj/item/organ/internal/heart/gland/proc/ownerCheck()
 	if(ishuman(owner))
-		return 1
+		return TRUE
 	if(!human_only && iscarbon(owner))
-		return 1
-	return 0
+		return TRUE
+	return FALSE
 
 /obj/item/organ/internal/heart/gland/proc/Start()
 	active = 1
 	next_activation = world.time + rand(cooldown_low,cooldown_high)
 
+/obj/item/organ/internal/heart/gland/proc/update_gland_hud()
+	if(!owner)
+		return
+	var/image/holder = owner.hud_list[GLAND_HUD]
+	var/icon/I = icon(owner.icon, owner.icon_state, owner.dir)
+	holder.pixel_y = I.Height() - world.icon_size
+	if(active_mind_control)
+		holder.icon_state = "hudgland_active"
+	else if(mind_control_uses)
+		holder.icon_state = "hudgland_ready"
+	else
+		holder.icon_state = "hudgland_spent"
+
+/obj/item/organ/internal/heart/gland/proc/mind_control(command, mob/living/user)
+	if(!ownerCheck() || !mind_control_uses || active_mind_control)
+		return
+	mind_control_uses--
+	to_chat(owner, "<span class='userdanger'>You suddenly feel an irresistible compulsion to follow an order...</span>")
+	to_chat(owner, "<span class='mind_control'>[command]</span>")
+	active_mind_control = TRUE
+	log_admin("[key_name(user)] sent an abductor mind control message to [key_name(owner)]: [command]")
+	update_gland_hud()
+
+	addtimer(CALLBACK(src, .proc/clear_mind_control), mind_control_duration)
+
+/obj/item/organ/internal/heart/gland/proc/clear_mind_control()
+	if(!ownerCheck() || !active_mind_control)
+		return
+	to_chat(owner, "<span class='userdanger'>You feel the compulsion fade, and you completely forget about your previous orders.</span>")
+	active_mind_control = FALSE
+	update_gland_hud()
 
 /obj/item/organ/internal/heart/gland/remove(var/mob/living/carbon/M, special = 0)
 	active = 0
 	if(initial(uses) == 1)
 		uses = initial(uses)
+	var/datum/atom_hud/abductor/hud = huds[DATA_HUD_ABDUCTOR]
+	hud.remove_from_hud(owner)
+	clear_mind_control()
 	. = ..()
 
 /obj/item/organ/internal/heart/gland/insert(var/mob/living/carbon/M, special = 0)
 	..()
 	if(special != 2 && uses) // Special 2 means abductor surgery
```

Comment:
We might want to have `#define`s for "special" surgery values, so we don't accidentally allow systems to step on each other's toes for values of special behavior (e.g. some other surgery/organs system implementing `special` as a bitfield or something, for controlling limb removal behavior)

Reply:
Probably so, though I feel this should probably be addressed in a separate PR.